WebDriverAgentLib/FBAlert.m

```diff
@@ -44,8 +44,16 @@ - (XCUIElement *)fb_alertElement
   if (alert.exists) {
     return alert;
   }
+    
+  //Only return sheet if it does not have a dismiss popover region.
+  BOOL isIpad = [UIDevice currentDevice].userInterfaceIdiom == UIUserInterfaceIdiomPad;
+  //check for Apple's popover region in a UIPopoverController.
+  NSPredicate *predicateString = [NSPredicate predicateWithFormat:@"identifier == 'PopoverDismissRegion'"];
+  XCUIElementQuery *query = [[self.application descendantsMatchingType:XCUIElementTypeAny] matchingPredicate:predicateString];
+  NSArray *childElements = [query allElementsBoundByIndex];
+    
   alert = self.sheets.element;
-  if (alert.exists) {
+  if ((alert.exists && isIpad && childElements.count == 0) || (alert.exists && !isIpad)) {
```

Comment:
This is hard to read what about `alert.exists && (!isIPad || childElements.count == 0)`?